src/VisualStudio/Core/Def/Implementation/ProjectSystem/CPS/CPSProject_IProjectContext.cs

```diff
@@ -103,15 +127,27 @@ public void AddProjectReference(IProjectContext project, MetadataReferenceProper
 
         public void RemoveProjectReference(IProjectContext project)
         {
-            foreach (var projectReference in GetCurrentProjectReferences())
+            var referencedProject = GetAbstractProject(project);
+
+            // AbstractProject and ProjectTracker track project references using the project bin output path.
+            // We fetch this output path to remove the reference.
+            var binPathOpt = referencedProject.TryGetBinOutputPath();
+            if (!string.IsNullOrEmpty(binPathOpt))
             {
-                if (projectReference.ProjectId.Equals(project.Id))
-                {
-                    RemoveProjectReference(projectReference);
-                    return;
-                }
+                base.RemoveMetadataReference(binPathOpt);
             }
         }
+
+        private AbstractProject GetAbstractProject(IProjectContext project)
```

Comment:
Should be static? Why have this versus just having callers explicitly cast to the underlying type?